frontend/src/tests/setup.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * @jest-environment node
+ */
+import { expect } from "./utils";
+
+class DatabaseSeeder {
+    constructor() {
+        this.seededData = {
+            session: {
+                start_date: new Date("2020-02-10").toISOString(),
+                end_date: new Date("2020-12-31").toISOString(),
+                name: "Initial Session",
+                rate1: 50
+            },
+            contractTemplate: {
+                template_name: "Regular",
+                template_file: "/path/to/regular/template.html"
+            },
+            position: {
+                position_code: "CSC494",
+                position_title: "Capstone Project",
+                hours_per_assignment: 20,
+                start_date: new Date("2020-01-01").toISOString(),
+                end_date: new Date("2020-05-01").toISOString(),
+                contract_template_id: null
+            },
+            applicant: {
+                utorid: "johnd",
+                student_number: "10000000",
+                first_name: "John",
+                last_name: "Doe",
+                email: "fake@email.com",
+                phone: "4166666666"
+            },
+            assignment: {
+                position_id: null,
+                applicant_id: null,
+                start_date: new Date("2020-01-01").toISOString(),
+                end_date: new Date("2020-05-01").toISOString()
+            }
+        };
+    }
+
+    /**
+     * Seeds the database by clearing it first, then seeding,
+     * then taking a snapshot.
+     *
+     * @param {*} api
+     * @memberof DatabaseSeeder
+     */
+    async seed(api) {
+        return await seedDatabase(api, this.seededData);
+    }
+
+    /**
+     * Verify that the data that was just seeded can be retrieved
+     * from the API.
+     *
+     * @param {*} api
+     * @memberof DatabaseSeeder
+     */
+    async verifySeed(api) {
+        return await verifySeededDatabase(api, this.seededData);
+    }
+}
+
+export const databaseSeeder = new DatabaseSeeder();
+
+/**
+ * Seeding the database with the minimal set of API calls to create an assignment.
+ */
+async function seedDatabase(
+    api,
+    seeded = {
+        session: null,
+        contractTemplate: null,
+        position: null,
+        applicant: null,
+        assignment: null
+    }
+) {
+    const { apiPOST } = api;
+
+    await apiPOST("/debug/clear_data");
+
+    //
+    // Seed the database with initial data. We assume that everything
+    // here will work without failing, so we won't work hard to verify the
+    // results.
+    //
+
+    let resp = null;
+
+    // Session
+    resp = await apiPOST("/admin/sessions", seeded.session);
+    // Save the data that was returned to us. What's most important
+    // is the session id. Use Object.assign so that all memory
+    // references to the original seeded.session are kept intact.
+    Object.assign(seeded.session, resp.payload);
```

Comment:
This probably does not happen a lot and sounds like nit picking, but should we change the second argument from `resp.payload` to `{id: respect.payload.id}` just to be extra cautious? With `Object.assign`, the second argument will overwrite the existing field in the first argument. I know that this rarely happens, but for the sake of testing, should we not assume that every field in the response is correct?

Reply:
I don't think so. We should prefer information from the backend. It might, for instance, reformat our dates in a particular way, and that would be fine.